### ISSUE TITLE
Unable to build a native image with GraalVM 20.2 and debug enabled

### ISSUE THREAD

**Issue by loicmathieu** [quarkusio/quarkus#12383]
**Describe the bug**
I have an application using Quarkus 1.8.1, it can successfully build a native image, but when I want to build one with debug enabled it fails.

Working native image creation command:
```
mvn clean package -DskipTests -Pnative
```

Failing native image creation command:
```
mvn clean package -DskipTests -Dquarkus.native.debug.enabled=true -Pnative
```

Native image exception:
```
Fatal error:com.oracle.svm.core.util.VMError$HostedError: com.oracle.svm.core.util.UserError$UserException: Image heap writing found a class not seen during static analysis. Did a static field or an object referenced from a static field change during native image generation? For example, a lazily initialized cache could have been initialized during image generation, in which case you need to force eager initialization of the cache before static analysis or reset the cache using a field value recomputation.
    object: 2014=[Ljava.time.zone.ZoneOffsetTransition;@6a44f804  of class: java.util.concurrent.ConcurrentHashMap$Node
  reachable through:
    object: [Ljava.util.concurrent.ConcurrentHashMap$Node;@1b47cdb0  of class: java.util.concurrent.ConcurrentHashMap$Node[]
    object: {2017=[Ljava.time.zone.ZoneOffsetTransition;@55ee4178, 2018=[Ljava.time.zone.ZoneOffsetTransition;@38c0d9c9, 2019=[Ljava.time.zone.ZoneOffsetTransition;@7cb691ca, 2020=[Ljava.time.zone.ZoneOffsetTransition;@f25316c, 2006=[Ljava.time.zone.ZoneOffsetTransition;@6b650fa4, 2013=[Ljava.time.zone.ZoneOffsetTransition;@4e85dc79, 2014=[Ljava.time.zone.ZoneOffsetTransition;@6a44f804}  of class: java.util.concurrent.ConcurrentHashMap
    object: ZoneRules[currentStandardOffset=+01:00]  of class: java.time.zone.ZoneRules
    object: Europe/Paris  of class: java.time.ZoneRegion
    object: sun.util.calendar.ZoneInfo[id="Europe/Paris",offset=3600000,dstSavings=3600000,useDaylight=true,transitions=184,lastRule=java.util.SimpleTimeZone[id=Europe/Paris,offset=3600000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=2,startMonth=2,startDay=-1,startDayOfWeek=1,startTime=3600000,startTimeMode=2,endMode=2,endMonth=9,endDay=-1,endDayOfWeek=1,endTime=3600000,endTimeMode=2]]  of class: sun.util.calendar.ZoneInfo
    object: 0000-12-30T16:47:04.192Z  of class: sun.util.calendar.ImmutableGregorianDate
    object: [CommonEra (C.E.) since 0000-12-30T16:47:04.192 local time]  of class: sun.util.calendar.Era
    object: [Lsun.util.calendar.Era;@2038004b  of class: sun.util.calendar.Era[]
    root: sun.util.calendar.JulianCalendar$Date.setNormalizedYear(int)

	at com.oracle.svm.core.util.VMError.shouldNotReachHere(VMError.java:72)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:661)
	at com.oracle.svm.hosted.NativeImageGenerator.lambda$run$0(NativeImageGenerator.java:468)
	at java.base/java.util.concurrent.ForkJoinTask$AdaptedRunnableAction.exec(ForkJoinTask.java:1407)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: com.oracle.svm.core.util.UserError$UserException: Image heap writing found a class not seen during static analysis. Did a static field or an object referenced from a static field change during native image generation? For example, a lazily initialized cache could have been initialized during image generation, in which case you need to force eager initialization of the cache before static analysis or reset the cache using a field value recomputation.
    object: 2014=[Ljava.time.zone.ZoneOffsetTransition;@6a44f804  of class: java.util.concurrent.ConcurrentHashMap$Node
  reachable through:
    object: [Ljava.util.concurrent.ConcurrentHashMap$Node;@1b47cdb0  of class: java.util.concurrent.ConcurrentHashMap$Node[]
    object: {2017=[Ljava.time.zone.ZoneOffsetTransition;@55ee4178, 2018=[Ljava.time.zone.ZoneOffsetTransition;@38c0d9c9, 2019=[Ljava.time.zone.ZoneOffsetTransition;@7cb691ca, 2020=[Ljava.time.zone.ZoneOffsetTransition;@f25316c, 2006=[Ljava.time.zone.ZoneOffsetTransition;@6b650fa4, 2013=[Ljava.time.zone.ZoneOffsetTransition;@4e85dc79, 2014=[Ljava.time.zone.ZoneOffsetTransition;@6a44f804}  of class: java.util.concurrent.ConcurrentHashMap
    object: ZoneRules[currentStandardOffset=+01:00]  of class: java.time.zone.ZoneRules
    object: Europe/Paris  of class: java.time.ZoneRegion
    object: sun.util.calendar.ZoneInfo[id="Europe/Paris",offset=3600000,dstSavings=3600000,useDaylight=true,transitions=184,lastRule=java.util.SimpleTimeZone[id=Europe/Paris,offset=3600000,dstSavings=3600000,useDaylight=true,startYear=0,startMode=2,startMonth=2,startDay=-1,startDayOfWeek=1,startTime=3600000,startTimeMode=2,endMode=2,endMonth=9,endDay=-1,endDayOfWeek=1,endTime=3600000,endTimeMode=2]]  of class: sun.util.calendar.ZoneInfo
    object: 0000-12-30T16:47:04.192Z  of class: sun.util.calendar.ImmutableGregorianDate
    object: [CommonEra (C.E.) since 0000-12-30T16:47:04.192 local time]  of class: sun.util.calendar.Era
    object: [Lsun.util.calendar.Era;@2038004b  of class: sun.util.calendar.Era[]
    root: sun.util.calendar.JulianCalendar$Date.setNormalizedYear(int)

	at com.oracle.svm.core.util.UserError.abort(UserError.java:68)
	at com.oracle.svm.hosted.image.NativeImageHeap.reportIllegalType(NativeImageHeap.java:508)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.verifyTargetDidNotChange(NativeImageHeapWriter.java:132)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.writeReference(NativeImageHeapWriter.java:165)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.write(NativeImageHeapWriter.java:154)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.writeConstant(NativeImageHeapWriter.java:191)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.writeObject(NativeImageHeapWriter.java:383)
	at com.oracle.svm.hosted.image.NativeImageHeapWriter.writeHeap(NativeImageHeapWriter.java:88)
	at com.oracle.svm.hosted.image.NativeBootImage.build(NativeBootImage.java:473)
	at com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:650)
	... 7 more
Error: Image build request failed with exit status 1

```

**Expected behavior**
Native image should be created when activating native image debug with Quarkus 1.8.1

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux 5.4.0-48-generic #52-Ubuntu SMP Thu Sep 10 10:58:49 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: openjdk version "11.0.7" 2020-04-14
 - GraalVM version (if different from Java): GraalVM CE 20.2.0 (build 11.0.8+10-jvmci-20.2-b03)
 - Quarkus version or git rev: 1.8.1.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 3.6.3

**Comment by zbendhiba** [comment]
I have the same issue this morning

**Comment by galderz** [comment]
Looks very similar to: https://github.com/oracle/graal/issues/2637

**Comment by galderz** [comment]
The fix didn't make it to GraalVM 20.2, but it is part of Mandrel 20.2, e.g.

```
./mvnw package -Pnative \
-Dquarkus.native.container-build=true \
-Dquarkus.native.builder-image=quay.io/quarkus/ubi-quarkus-mandrel:20.2.0.0.Beta1-java11
```

**Comment by galderz** [comment]
See [Mandrel release notes](https://github.com/graalvm/mandrel/releases/tag/mandrel-20.2.0.0.Beta1) for advanced details on how you can try it bare metal on Linux.

**Comment by loicmathieu** [comment]
It works using Mandrel 20.2.
It takes more heap to build the native executable but then I'm able to use gbg.
I close the issue as it's fixed in the upcomming Mandrel 20.2 and GraalVM 20.3 releases.
